libdnf/sack/query.cpp

```diff
@@ -1915,12 +1925,16 @@ Query::Impl::filterAdvisory(const Filter & f, Map *m, int keyname)
             }
 
             Id id = -1;
-            // Add to candidates resultPset pkgs that match name and arch with some already installed pkg
+            // Add to candidates resultPset pkgs that match name and arch with some already installed pkg or match name and either the installed or candidate are NOARCH
             while ((id = resultPset->next(id)) != -1) {
                 Solvable * s = pool_id2solvable(pool, id);
-                auto low = std::lower_bound(installed_solvables.begin(), installed_solvables.end(), s, NameArchSolvableComparator);
-                if (low != installed_solvables.end() && s->name == (*low)->name && s->arch == (*low)->arch) {
-                    candidates.push_back(s);
+                auto low = std::lower_bound(installed_solvables.begin(), installed_solvables.end(), s, NameSolvableComparator);
+                while (low != installed_solvables.end() && (*low)->name == s->name) {
+                    if (s->arch == (*low)->arch || s->arch == ARCH_NOARCH || (*low)->arch == ARCH_NOARCH) {
+                        candidates.push_back(s);
+                        break;
+                    }
+                    low++;
```

Comment:
I am sorry, I overlooked one thing - minor - `++low` might be a better.I guess that compilator will  fix that. What's the difference between `++low` and `low++`. In both approaches low iterator jump to the next element, but `low++` requires more resources because it returns iterator at current position and then move low to next item.

Reply:
Oh interesting.
I have updated it.